File: C++/Ders/Ders-2/Student.cpp
#include "Student.h"

Student::Student(std::string name, std::string surname, int age, int ID) {
	this->name = name;
	this->surname = surname;
	this->age = age;
	this->ID = ID;
	
}
void Student::print() {
	std::cout << "Name: " << this->name
		<< "\nSurname: " << this->surname 
		<< "\nID: " << this->ID << "\nAge: " 
		<< this->age << std::endl;
}

File: JAVA/Ders/Ders-5/src/ExceptionHandling/Student.java
package ExceptionHandling;

public class Student {
    String name, surname;
    int ID;

    public Student(String name, String surname, int ID) throws InvalidIDException {
        this.name = name;
        this.surname = surname;
        if(ID > 0)
            this.ID = ID;
        else
            throw new InvalidIDException("ID cannot be < 0");
    }

    @Override
    public String toString() {
        return "Student{" +
                "name='" + name + '\'' +
                ", surname='" + surname + '\'' +
                ", ID=" + ID +
                '}';
    }
}
